File: src/0104_exploratory_data_analysis/05_data_visualisation/script_03.R

jobs_data <- read.csv("./data/0104_exploratory_data_analysis/data_visualisation/JOB\ PROFICIENCY\ DATA.csv", header = TRUE)
attach(jobs_data)
head(jobs_data)

plot(
  aptitude,
  job_prof,
  main = "Scatterplot with Regression Line",
  xlab = "Aptitude",
  ylab = "Job Proficiency",
  pch = 19
)
abline(lm(job_prof ~ aptitude), col = "darkorange")

pairs(
  ~ aptitude + job_prof + testofen + tech_ + g_k_,
  data = jobs_data,
  main = "Scatterplot Matrix",
  col = "darkorange"
)




library(GGally)
ggpairs(jobs_data[, c("aptitude", "testofen", "tech_", "g_k_", "job_prof")], title = "Scatter Plot Matrix")




library(ggplot2)
qplot(
  x = tech_, 
  y = job_prof, 
  data = jobs_data, 
  color = aptitude, 
  size = aptitude, 
  xlab = "Technical", 
  ylab = "Job Proficiency", 
  main = "Bubble Chart"
)

File: scripts/0104_exploratory_data_analysis/05_data_visualisation/script_03.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Thu Oct  5 13:58:53 2023

@author: jerry
"""


# %%

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns



# %%

employee_data = pd.read_csv("./data/0104_exploratory_data_analysis/05_data_visualisation/JOB PROFICIENCY DATA.csv", index_col = 0)
employee_data.info()



# %%

plt.figure()
sns.lmplot('aptitude', 'job_prof', data = employee_data)
plt.title('Scatter Plot with Regression Line')
plt.xlabel('Aptitude')
plt.ylabel('Job Proficiency')



# %%

plt.figure()
sns.pairplot(employee_data)
plt.title('Scatter Plot Matrix')



# %%

plt.figure()
sns.scatterplot('tech_', 'job_prof', data = employee_data, hue = 'aptitude', size = 'aptitude')
plt.title('Bubble Chart')
plt.xlabel('Technical')
plt.ylabel('Job Proficiency')
